### PR TITLE
Added definition for redact-axios-error

### PR DESCRIPTION
**Pull request by AminFazlMondo**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/redact-axios-error/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// Type definitions for redact-axios-error 4.0
+// Project: https://github.com/tcollinsworth/redact-axios-error#readme
+// Definitions by: Amin Fazl <https://github.com/me>
```
Comment (sandersn) [inline types/redact-axios-error/index.d.ts:3]: Please put your github user name in the url

**Review by sandersn — CHANGES_REQUESTED** [review]
Looks good except for the github url

**Review by peterblazejewicz — COMMENTED** [review]
@AminFazlMondo do you plan to support direct imports as outlined in project readme?
https://github.com/tcollinsworth/redact-axios-error#interceptor-usage

so in your tests this would be:
```ts
import { AxiosErrorGroomer } from 'redact-axios-error';
import { getAxiosErrorInterceptor } from 'redact-axios-error/axios-error-redact-interceptor';

getAxiosErrorInterceptor(new AxiosErrorGroomer());
```
to support those imports you would have to split declaratins to align with native modules, so:

```ts
// axios-error-redact-interceptor.d.ts

import { AxiosErrorGroomer } from "./";

export function getAxiosErrorInterceptor(axiosErrorGroomer?: AxiosErrorGroomer | null): (error: any) => Promise<never>;
```

**Comment by AminFazlMondo** [comment]
Thanks, @peterblazejewicz,
I have updated the code, the code supports importing from index or actual module, not sure how to write test for both.I have updated the code to be modular though.

File: types/redact-axios-error/axios-error-groomer.d.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  /**
+   * @param includeRequestData default true, false redacts request data
+   * @param includeResponseData default true, false redacts response data
+   * @param includeQueryData default true, false redacts query string data
+   */
+  constructor(includeRequestData?: boolean, includeResponseData?: boolean, includeQueryData?: boolean);
+
+  includeRequestData(flag: boolean): void;
+  includeResponseData(flag: boolean): void;
+  includeQueryData(flag: boolean): void;
```
Comment (peterblazejewicz) [inline types/redact-axios-error/axios-error-groomer.d.ts:11]:
```suggestion
  includeRequestData(flag: boolean): this;
  includeResponseData(flag: boolean): this;
  includeQueryData(flag: boolean): this;
```

**Comment by peterblazejewicz** [comment]
Yuo're missing public method:
```ts
isAxiosError(err: any): err is Error;
```

File: types/redact-axios-error/axios-error-groomer.d.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  getGroomedAxiosError(err: any): any;
```
Comment (peterblazejewicz) [inline types/redact-axios-error/axios-error-groomer.d.ts:14]:
```suggestion
  getGroomedAxiosError(err: Error | null | undefined): Error | null | undefined;
```
https://github.com/tcollinsworth/redact-axios-error/blob/master/redact-axios-error.js#L29
you can probably do that way. Please verify with your Axios code if this looks OK.

**Review by RyanCavanaugh — APPROVED** [review]

**Comment by AminFazlMondo** [comment]
Ready to merge

**Pull request merged**